### PR TITLE
Implemented Red Alert 2 tesla bolt engine effects

### PR DESCRIPTION
**Pull request by Mailaender**
and removed Red Alert 1 tesla bolt sprites.

### CODE REVIEW

File: OpenRA.Mods.RA2/Projectiles/ElectricBolt.cs [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+		readonly WVec upVector;
+		readonly MersenneTwister random;
+		readonly bool hasLaunchEffect;
+
+		[Sync]
+		readonly WPos target, source;
+
+		int ticks = 0;
+
+		HashSet<Pair<Color, WPos[]>> zaps;
```
Comment (abcdefg30) [inline OpenRA.Mods.RA2/Projectiles/ElectricBolt.cs]: This can be readonly?

File: OpenRA.Mods.RA2/Projectiles/ElectricBolt.cs [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+		public ElectricBolt(ElectricBoltInfo info, ProjectileArgs args)
+		{
+			this.args = args;
+			this.info = info;
+			var playerColors = args.SourceActor.Owner.Color;
+			var colors = info.Colors;
+			for (int i = 0; i < info.PlayerColorZaps; i++)
```
Comment (abcdefg30) [inline OpenRA.Mods.RA2/Projectiles/ElectricBolt.cs]: var

File: OpenRA.Mods.RA2/Projectiles/ElectricBolt.cs [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+				colors.Append(playerColors);
+
+			target = args.PassiveTarget;
+			source = args.Source;
+			random = args.SourceActor.World.LocalRandom;
```
Comment (Mailaender) [inline OpenRA.Mods.RA2/Projectiles/ElectricBolt.cs]: I am not sure if caching this is a good idea.

**Comment by sorcerer86pt** [comment]
lgtm:

https://drive.google.com/file/d/1zPU5ePq1r2KzEQaLvblOOB8GLOP1YqVn/view?usp=sharing

**Pull request merged**